linux/keyman-config/keyman_config/uninstall_kmp.py

```diff
@@ -65,18 +69,36 @@ def uninstall_keyboards_from_ibus(keyboards, packageDir):
     if bus:
         # install all kmx for first lang not just packageID
         for kb in keyboards:
-            kmx_file = os.path.join(packageDir, kb['id'] + ".kmx")
-            if "languages" in kb and len(kb["languages"]) > 0:
-                logging.debug(kb["languages"][0])
-                keyboard_id = "%s:%s" % (kb["languages"][0]['id'], kmx_file)
-            else:
-                keyboard_id = kmx_file
+            keyboard_id = get_keyboard_id(kb, packageDir)
             uninstall_from_ibus(bus, keyboard_id)
         restart_ibus(bus)
     else:
         logging.warning("could not uninstall keyboards from IBus")
 
 
+def uninstall_keyboards_from_gnome(keyboards, packageDir):
+    gnomeKeyboardsUtil = GnomeKeyboardsUtil()
+    sources = gnomeKeyboardsUtil.read_input_sources()
+
+    # uninstall all kmx for all languages
+    for kb in keyboards:
+        keyboard_id = get_keyboard_id(kb, packageDir)
```

Comment:
Please rename `keyboard_id` to something that qualifies it as a `gnome_keyboard_id`?

Reply:
done. Named it `ibus_keyboard_is`.